### PR TITLE
Shuffle Service

### PR DESCRIPTION
**Pull request by mrocklin**
- [ ] Closes #xxxx
- [ ] Tests added / passed
- [ ] Passes `pre-commit run --all-files`

### CODE REVIEW

File: dask/dataframe/shuffle.py [hunk 1]
```diff
@@ -495,6 +495,10 @@ def rearrange_by_column(
         if ignore_index:
             df2._meta = df2._meta.reset_index(drop=True)
         return df2
+    elif shuffle == "p2p":
```
Comment (hayesgb) [inline dask/dataframe/shuffle.py:498]: Suggest this be wrapped in a `try / except` block.  Could make a case for including a warning about the experimental nature.

File: dask/dataframe/shuffle.py [hunk 2]
```diff
@@ -495,6 +495,10 @@ def rearrange_by_column(
...
+        from distributed.shuffle import rearrange_by_column_p2p
```
Comment (hayesgb) [inline dask/dataframe/shuffle.py:499]: https://github.com/dask/distributed/issues/5996#issuecomment-1078245372

Comment (mrocklin) [reply dask/dataframe/shuffle.py:499]: This has been in distributed since the skeleton was first merged.  We're safe here on import.

**Comment by mrocklin** [comment]
Removing WIP.  I'd like to merge this in 24 hours if there are no further comments.

**Pull request merged**